### PR TITLE
update rbac for searchschema and searchcomplete

### PR DESCRIPTION
**Pull request by SherinV**
Signed-off-by: Sherin Varughese <shvarugh@redhat.com>

**Related Issue:**  https://github.com/stolostron/backlog/issues/21298

### Description of changes
Added  a map to lookup cluster scoped resources - to exclude them from namespace scoped filter.
Updated schema and searchcomplete with rbac clause.

### CODE REVIEW

File: pkg/rbac/userData.go
Comment (jlpadilla) [inline pkg/rbac/userData.go:132]: Include the user id in case we need to debug multiple users.

Comment (SherinV) [reply pkg/rbac/userData.go]: Added

**Comment by SherinV** [comment]
PR under review

File: pkg/rbac/sharedData.go
Comment (jlpadilla) [inline pkg/rbac/sharedData.go:20]: Wonder if we should avoid duplication and remove `csResources` and only use `csResourceMap` in all places.

File: pkg/rbac/sharedData.go
Comment (jlpadilla) [inline pkg/rbac/sharedData.go:104]: Try to keep these consistent, use either `nil or make(...)` on both to avoid confusion.
Would be better if we can only use the map.

File: pkg/rbac/userData.go
Comment (jlpadilla) [inline pkg/rbac/userData.go:120]: Are we able to proceed without the `uid` or should we return an error here?

File: pkg/rbac/userData.go
Comment (jlpadilla) [inline pkg/rbac/userData.go:272]: 🤔 I wonder if it is possible to create a cluster-scoped CR with the same group and name as a namespaced CR.
This would definitely be an edge scenario.

File: pkg/resolver/searchComplete.go
Comment (jlpadilla) [inline pkg/resolver/searchComplete.go:89]: Do we have any context that we could add here to help us debug if we panic here? I'm thinking something like the user ID, the input query, or anything else that could help us later.

File: pkg/resolver/searchComplete.go
Comment (jlpadilla) [inline pkg/resolver/searchComplete.go:144]: I got confused by the scope of `isNumber`, could you use different variable names?
```suggestion
		isNumber := isNumber(srchCompleteOut)
		if isNumber { //check if valid number
			isNumberStr := "isNumber"
			srchCompleteOutNum := []*string{&isNumberStr} //isNumber should be the first argument if the property is a number
```

File: pkg/resolver/searchComplete.go
Comment (jlpadilla) [inline pkg/resolver/searchComplete.go:162]: And please variables to match my previous comment.
```suggestion
		if !isNumber && isDate(srchCompleteOut) { //check if valid date
			isDateStr := "isDate"
			srchCompleteOutDate := []*string{&isDateStr}
			srchCompleteOut = srchCompleteOutDate
```

File: pkg/rbac/userData.go
Comment (SherinV) [reply pkg/rbac/userData.go:120]: This uid is being used for logging only. Moved it.

File: pkg/rbac/sharedData.go
Comment (SherinV) [reply pkg/rbac/sharedData.go:20]: Will move in next PR

File: pkg/resolver/searchComplete.go
Comment (jlpadilla) [inline pkg/resolver/searchComplete.go:121]: This looks like debug code. We could use a flag like `DEBUG_QUERY` if this can be helpful in the future.

File: pkg/resolver/searchComplete.go
Comment (jlpadilla) [inline pkg/resolver/searchComplete.go:146]: Please change to klog or delete.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

Please remove or adjust debugging code and logs before merging.

**Pull request merged**
